### PR TITLE
add a version flag to the release tool

### PR DESCRIPTION
**Pull request by caesarxuchao**
My code in the release tool had wrong assumption that outputDir == release version, which is not true. This PR explicitly adds a release version flag.

@nikhiljindal @RichieEscarez

### CODE REVIEW

File: _tools/release_docs/main.go [hunk 1]
```diff
@@ -36,6 +36,7 @@ import (
 var (
 	branch       = flag.String("branch", "", "The git branch from which to pull docs. (e.g. release-1.0, master).")
 	outputDir    = flag.String("output-dir", "", "The directory in which to save results.")
+	version      = flag.String("version", "", "The release version. The generated webpages will be at k8s.io/$version.")
```
Comment (nikhiljindal) [inline _tools/release_docs/main.go]: Give an example

File: _tools/release_docs/main.go [hunk 2]
```diff
@@ -431,7 +432,7 @@ func main() {
 
 			if *apiReference && !info.IsDir() && (info.Name() == "definitions.html" || info.Name() == "operations.html") {
 				fmt.Printf("Processing %s\n", path)
-				err := processHTML(path, *outputDir)
+				err := processHTML(path, *version, *outputDir)
```
Comment (nikhiljindal) [inline _tools/release_docs/main.go]: You dont check anywhere that it is non-empty?

**Comment by caesarxuchao** [comment]
@nikhiljindal  fixed. PTAL. Thank you!

File: _tools/release_docs/main.go [hunk 3]
```diff
@@ -36,6 +36,7 @@ import (
 var (
 	branch       = flag.String("branch", "", "The git branch from which to pull docs. (e.g. release-1.0, master).")
 	outputDir    = flag.String("output-dir", "", "The directory in which to save results.")
+	version      = flag.String("version", "", "The release version. We assume this version will be relected in the generated pages' URL, e.g., we assume for release v1.0, its webpages will be hosted at k8s.io/v1.0.")
```
Comment (nikhiljindal) [inline _tools/release_docs/main.go]: relected?
It is still not clear, whats a valid value for this. 1.0? v1.0? k8s.io/v1.0?

Comment (nikhiljindal) [inline _tools/release_docs/main.go]: While you are at it, will be great if you add an example for output-dir as well.

Comment (nikhiljindal) [inline _tools/release_docs/main.go]: It might also be good to mention that it is a required flag. (Same for output-dir)

File: _tools/release_docs/main.go [hunk 4]
```diff
@@ -396,6 +397,10 @@ func main() {
 		fmt.Println("You must specify an output dir with --output-dir.")
 		os.Exit(1)
 	}
+	if len(*version) == 0 {
```
Comment (nikhiljindal) [inline _tools/release_docs/main.go]: optional: We should give out all errors together, rather than one at a time :)

File: _tools/release_docs/main.go (hunk 3 above)
Comment (caesarxuchao) [inline _tools/release_docs/main.go]: Oh, I see your point now. Will fix.

Comment (caesarxuchao) [inline _tools/release_docs/main.go]: output-dir is used here:
https://github.com/kubernetes/kubernetes/blob/gh-pages/_tools/import_docs.sh#L23

It's actually set to a temp dir.

File: _tools/release_docs/main.go (hunk 4 above)
Comment (caesarxuchao) [inline _tools/release_docs/main.go]: You are right. Let's improve this in another PR :)

File: _tools/release_docs/main.go [hunk 5]
```diff
@@ -36,8 +36,9 @@ import (
 var (
 	branch       = flag.String("branch", "", "The git branch from which to pull docs. (e.g. release-1.0, master).")
 	outputDir    = flag.String("output-dir", "", "The directory in which to save results.")
-	remote       = flag.String("remote", "upstream", "The name of the remote repo from which to pull docs.")
-	apiReference = flag.Bool("apiReference", true, "Whether update api reference")
+	version      = flag.String("version", "", "The release version. It should be the same as the version segment in the URL, e.g., release v1.0 is hosted at k8s.io/v1.0/, so the version flag should be \"v1.0\".")
```
Comment (caesarxuchao) [inline _tools/release_docs/main.go]: @nikhiljindal How about this?

**Comment by caesarxuchao** [comment]
Ping @nikhiljindal :)

We need this to produce correct links.

File: _tools/import_docs.sh [hunk 6]
```diff
@@ -20,7 +20,7 @@ tmpdir=docs.$RANDOM
 echo fetching upstream
 git fetch upstream
 go build ./_tools/release_docs
-./release_docs --branch ${BRANCH} --output-dir $tmpdir >/dev/null
+./release_docs --branch ${BRANCH} --output-dir $tmpdir --version $OUTDIR >/dev/null
```
Comment (krousey) [inline _tools/import_docs.sh]: This seems to still be confusing versions and paths

Comment (caesarxuchao) [inline _tools/import_docs.sh]: Updated. PTAL.

File: _tools/import_docs.sh [hunk 7]
```diff
@@ -1,6 +1,7 @@
 #!/bin/sh
 # This is a hack to import from a branch and call it a versioned snapshot.
 
+VERSION=v1.0
```
Comment (nikhiljindal) [inline _tools/import_docs.sh]: As discussed, there will be another PR changing these to v1.1

File: _tools/release_docs/main.go (hunk 5 above)
Comment (nikhiljindal) [inline _tools/release_docs/main.go]: can change it to "for ex: for docs at k8s.io/v1.0, value of this flag should be v1.0"

**Comment by caesarxuchao** [comment]
@nikhiljindal update the flag description as discussed. PTAL. Thanks!

**Comment by nikhiljindal** [comment]
LGTM, thanks!

File: _tools/release_docs/api-reference-process.go [hunk 8]
```diff
@@ -67,9 +67,9 @@ func stripHTML(filename string) error {
 
 // replace http://releases.k8s.io/HEAD/docs/xxx.md with
 // http://kubernetes.io/v1.0/docs/xxx.html
-func fixLinks(fileName, outputDir string) error {
+func fixLinks(fileName, version string) error {
 	mdRE := regexp.MustCompile(`http://releases.k8s.io/HEAD/docs(.*?\.)md`)
-	repl := []byte("http://kubernetes.io/" + path.Base(outputDir) + "/docs" + "${1}" + "html")
+	repl := []byte("http://kubernetes.io/" + path.Base(version) + "/docs" + "${1}" + "html")
```
Comment (thockin) [inline _tools/release_docs/api-reference-process.go]: should not need path.Base on this

Comment (caesarxuchao) [inline _tools/release_docs/api-reference-process.go]: Yes. Fixed. Thanks. Can we merge manually?

**Comment by eparis** [comment]
So the bot seems to be getting angry here because this PR is against the gh-pages branch, not one that travis, shippable, etc pay attention to. I'll try to figure out how to shut up the bot (or maybe you meant this to be against a different branch?)

**Comment by caesarxuchao** [comment]
I'm going to self merge this since @nikhiljindal "lgtm"ed it and he will not be available in the next a few hours. I have more PRs depending on this one.

File: _tools/release_docs/main.go [hunk 9]
```diff
@@ -36,8 +36,9 @@ import (
 var (
 	branch       = flag.String("branch", "", "The git branch from which to pull docs. (e.g. release-1.0, master).")
 	outputDir    = flag.String("output-dir", "", "The directory in which to save results.")
-	remote       = flag.String("remote", "upstream", "The name of the remote repo from which to pull docs.")
-	apiReference = flag.Bool("apiReference", true, "Whether update api reference")
+	version      = flag.String("version", "", "The release version. It should be the same as the version segment in the URL, e.g., when importing docs that will be hosted at k8s.io/v1.0/, the version flag should be \"v1.0\".")
```
Comment (nikhiljindal) [inline _tools/release_docs/main.go]: Sorry I just noticed, we should have updated the "usage:" line at the top (line 18)

Comment (caesarxuchao) [inline _tools/release_docs/main.go]: submitted #16206 :)
